src/fingerprint.rs

```diff
@@ -10,8 +10,20 @@ const ZIP32_SEED_FP_PERSONALIZATION: &[u8; 16] = b"Zcash_HD_Seed_FP";
 /// The fingerprint for a wallet's seed bytes, as defined in [ZIP 32].
 ///
 /// [ZIP 32]: https://zips.z.cash/zip-0032#seed-fingerprints
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct SeedFingerprint([u8; 32]);
 
+impl ::core::fmt::Debug for SeedFingerprint {
+    fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
+        write!(f, "0x")?;
+        for i in self.0 {
+            write!(f, "{:02x}", i)?;
+        }
+
```

Comment:
I guess there's no point in depending on `hex` just for this. If we did it would be:
```rust
        write!(f, "0x{}", hex::encode(self.0))?;
```